openlibrary-wikidata-bot/jobs/sync_author_identifiers_from_wikidata.py

```diff
@@ -0,0 +1,210 @@
+from datetime import datetime
+from olclient import OpenLibrary
+from os import makedirs
+import argparse
+import csv
+import datetime
+import json
+import logging
+import re
+import sys
+
+# Setup logger
+logger = logging.getLogger("jobs.sync_author_wikidata_ids")
+logger.setLevel(logging.DEBUG)
+# The logging format
+log_formatter = logging.Formatter('%(name)s;%(levelname)-8s;%(asctime)s %(message)s')
+# Log warnings+ to console
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.WARN)
+console_handler.setFormatter(log_formatter)
+logger.addHandler(console_handler)
+# Log INFO+ to the log file
+log_dir = 'logs/jobs/sync_author_wikidata_ids'
+problem_dir = 'exports/jobs/sync_author_wikidata_ids'
+makedirs(log_dir, exist_ok=True)
+makedirs(problem_dir, exist_ok=True)
+log_file_datetime = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+log_file = log_dir + '/sync_author_wikidata_ids_%s.log' % log_file_datetime
+file_handler = logging.FileHandler(log_file)
+file_handler.setLevel(logging.DEBUG)
+file_handler.setFormatter(log_formatter)
+logger.addHandler(file_handler)
+
+# Setup error sheet
+n = datetime.datetime.now().strftime("%Y-%m-%d-%H%M%S")
+csv_file_path = f'{problem_dir}/sync_author_wikidata_ids_merge_problems-{n}.csv'
+with open(csv_file_path, mode="w", newline="", encoding="utf-8") as file:
+    writer = csv.writer(file)
+    writer.writerow(["wdid", "olid", "author_name", "problem", "identifier", "details"])  # Write header
+
+# I'd like it if this could come from identifiers.yml, but that's in a totally different repo!
+# In an ideal world, identifiers.yml stores the WD P### identifier for each remote ID type, and we can loop thru that instead.
+# Wikidata JSONs only use the P### identifier, so that's our only way to target specific remote IDs.
+WD_IDENTIFIERS = {
```

Comment:
Useful to note what this is mapping from/to:

```suggestion
WD_PID_TO_OL_IDENTIFIER_NAME = {
```